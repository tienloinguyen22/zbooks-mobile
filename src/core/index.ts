export * from './i18n';
export * from './themes/colors';
export * from './interfaces/ScreenProps';
export * from './interfaces/LoginType';
export * from './interfaces/User';
export * from './interfaces/Resource';
export * from './interfaces/Theme';
export * from './helpers/show_notification';
export * from './helpers/sleep';
export * from './helpers/record_error';
export * from './helpers/configure_google_sign_in';
export * from './helpers/combine_styles';
export * from './helpers/configure_google_sign_in';
export * from './helpers/handle_error';
export * from './helpers/get_primary_color';
export * from './helpers/format_date';
export * from './screen_names';
