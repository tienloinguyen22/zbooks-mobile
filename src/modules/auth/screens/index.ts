export * from './LoginScreen';
export * from './AppLoaderScreen';
export * from './FinishRegisterScreen';
