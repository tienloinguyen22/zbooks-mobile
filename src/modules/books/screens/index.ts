export * from './FavoriteBooksScreen';
export * from './BookDetailScreen';
