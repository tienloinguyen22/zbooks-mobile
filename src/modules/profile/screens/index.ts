export * from './UserProfileScreen';
export * from './EditEmailScreen';
export * from './EditFullnameScreen';
export * from './EditPhoneNoScreen';
