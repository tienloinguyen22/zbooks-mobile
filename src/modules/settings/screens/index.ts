export * from './SettingsScreen';
export * from './CategoryPreferenceScreen';
