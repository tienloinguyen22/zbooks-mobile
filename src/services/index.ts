export * from './app_service';
export * from './auth_service';
export * from './navigation_service';
