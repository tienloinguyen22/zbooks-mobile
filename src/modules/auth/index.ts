import { WithScreen, WithApolloClient } from '@app/components';
import { Navigation } from 'react-native-navigation';
import { screenNames } from '@app/core';
import { LoginScreen, AppLoaderScreen, FinishRegisterScreen } from './screens';

const registerScreens = (): void => {
  Navigation.registerComponent(screenNames.LoginScreen, () =>
    WithScreen(WithApolloClient(LoginScreen), {
      orientation: 'PORTRAIT',
    }),
  );

  Navigation.registerComponent(screenNames.AppLoaderScreen, () =>
    WithScreen(WithApolloClient(AppLoaderScreen), {
      orientation: 'PORTRAIT',
    }),
  );

  Navigation.registerComponent(screenNames.FinishRegisterScreen, () =>
    WithScreen(WithApolloClient(FinishRegisterScreen), {
      orientation: 'PORTRAIT',
    }),
  );
};

export default {
  registerScreens,
};
