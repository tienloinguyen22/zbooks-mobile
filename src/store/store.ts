import { connect as reduxConnect, ConnectedComponent } from 'react-redux';
import createRematchPersist from '@rematch/persist';
import { init, RematchRootState, RematchDispatch } from '@rematch/core';
import storage from '@react-native-community/async-storage';
import { persistStore } from 'redux-persist';
import { models } from './models';

const persistPlugin = createRematchPersist({
  whitelist: ['currentUser', 'settings', 'sharks'],
  throttle: 1000,
  version: 1,
  storage,
});

export const store = init({
  plugins: [persistPlugin],
  models,
});

export const persistor = persistStore(store, undefined, () => {
  const { settings } = store.getState();
  // we only should do it only 1 time when user installs & opens the app for the first time
  !settings.appLoaded && store.dispatch.settings.finishLoadingApp();
});

export type Store = typeof store;
export type Dispatch = RematchDispatch<typeof models>;
export type RootState = RematchRootState<typeof models>;

type Connect = (
  mapStateToProps: (state: RootState) => {},
  mapDispatchToProps: (dispatch: Dispatch) => {},
) => // eslint-disable-next-line @typescript-eslint/no-explicit-any
(component: any) => ConnectedComponent<any, any>;

export const connect: Connect = reduxConnect;
