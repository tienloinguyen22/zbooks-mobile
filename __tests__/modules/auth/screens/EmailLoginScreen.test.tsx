import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';
import { navigationService, authService } from '@app/services';
import { store, Dispatch, RootState } from '@app/store';
import { THEME_DARK, screenNames, sleep, ErrorWithCode, User } from '@app/core';
import * as handleError from '@app/core/helpers/handle_error';
import { EmailLoginScreen } from '@app/modules/auth/screens';
import { mockTheme } from '../../../test_helpers';

describe('modules/auth/screen/EmailLoginScreen', () => {
  const componentId = 'componentId';
  const email = 'email@email.com';
  const password = 'Password@12345';
  beforeEach(() => {
    mockTheme(THEME_DARK);
    navigationService.navigateTo = jest.fn();
    navigationService.setRootEmailVerification = jest.fn();
    store.getState = () =>
      (({
        settings: {
          language: 'en',
        },
      } as unknown) as RootState);
    store.dispatch = ({
      currentUser: {
        login: jest.fn(),
      },
    } as unknown) as Dispatch;
  });

  it('renders successfully', async () => {
    const { baseElement } = render(<EmailLoginScreen componentId={componentId} />);

    expect(baseElement).toMatchSnapshot();
  });

  it('renders successfully with Vietnamese language', async () => {
    store.getState = () =>
      (({
        settings: {
          language: 'vi',
        },
      } as unknown) as RootState);

    const { baseElement } = render(<EmailLoginScreen componentId={componentId} />);

    expect(baseElement).toMatchSnapshot();
  });

  it('navigates to Forgot Password screen after Forgot Password button is pressed', async () => {
    const { getByText } = render(<EmailLoginScreen componentId={componentId} />);
    const buttonForgotPassword = getByText('emailLoginScreen.forgotPassword');
    fireEvent.press(buttonForgotPassword);

    expect(navigationService.navigateTo).toBeCalledWith({
      componentId,
      screenName: screenNames.ForgotPasswordScreen,
    });
  });

  it('navigates to home when user logins successfully', async () => {
    const user: User = {
      id: 'id',
      avatarUrl: 'avatarUrl',
      isLoggedIn: true,
      loginType: 'EMAIL',
      emailVerified: true,
    };
    jest.spyOn(authService, 'signInWithEmailAndPassword').mockImplementation(async () => user);
    jest.spyOn(navigationService, 'setRootHome').mockImplementation(async () => {});

    const { getByPlaceholderText, getAllByText } = render(<EmailLoginScreen componentId={componentId} />);
    const buttonLogin = getAllByText('emailLoginScreen.login')[1];
    const fieldEmail = getByPlaceholderText('emailLoginScreen.email');
    const fieldPassword = getByPlaceholderText('emailLoginScreen.password');
    fireEvent.changeText(fieldEmail, email);
    fireEvent.changeText(fieldPassword, password);
    fireEvent.press(buttonLogin);

    // wait for event triggered by Formik
    await sleep(1000);

    expect(authService.signInWithEmailAndPassword).toBeCalledWith(email, password);
    expect(store.dispatch.currentUser.login).toBeCalledWith(user);
    expect(navigationService.setRootHome).toBeCalledTimes(1);
  });

  it('navigates to Email Verification screen when user logins successfully but his email is not verified', async () => {
    jest.spyOn(authService, 'signInWithEmailAndPassword').mockImplementation(async () => ({
      id: 'id',
      avatarUrl: 'avatarUrl',
      isLoggedIn: true,
      loginType: 'EMAIL',
      emailVerified: false,
    }));

    const { getByPlaceholderText, getAllByText } = render(<EmailLoginScreen componentId={componentId} />);
    const buttonLogin = getAllByText('emailLoginScreen.login')[1];
    const fieldEmail = getByPlaceholderText('emailLoginScreen.email');
    const fieldPassword = getByPlaceholderText('emailLoginScreen.password');
    fireEvent.changeText(fieldEmail, email);
    fireEvent.changeText(fieldPassword, password);
    fireEvent.press(buttonLogin);

    // wait for event triggered by Formik
    await sleep(1000);

    expect(authService.signInWithEmailAndPassword).toBeCalledWith(email, password);
    expect(navigationService.setRootEmailVerification).toBeCalledTimes(1);
  });

  it('shows warnings if wrong password is inputted', async () => {
    jest.spyOn(authService, 'signInWithEmailAndPassword').mockImplementation(async () => {
      const error = new Error() as ErrorWithCode;
      error.code = 'auth/wrong-password';
      throw error;
    });
    jest.spyOn(handleError, 'handleError').mockImplementation(async () => {});

    const { getByPlaceholderText, getAllByText } = render(<EmailLoginScreen componentId={componentId} />);
    const buttonLogin = getAllByText('emailLoginScreen.login')[1];
    const fieldEmail = getByPlaceholderText('emailLoginScreen.email');
    const fieldPassword = getByPlaceholderText('emailLoginScreen.password');
    fireEvent.changeText(fieldEmail, email);
    fireEvent.changeText(fieldPassword, password);
    fireEvent.press(buttonLogin);

    // wait for event triggered by Formik
    await sleep(1000);

    expect(handleError.handleError).toBeCalled();
  });

  it('shows warnings if information is inputted incorrectly', async () => {
    jest.spyOn(authService, 'signInWithEmailAndPassword').mockImplementation(async () => {
      const error = new Error() as ErrorWithCode;
      error.code = 'auth/wrong-password';
      throw error;
    });
    jest.spyOn(handleError, 'handleError').mockImplementation(async () => {});

    const { getByPlaceholderText, getAllByText, getByText } = render(<EmailLoginScreen componentId={componentId} />);
    const buttonLogin = getAllByText('emailLoginScreen.login')[1];
    const fieldEmail = getByPlaceholderText('emailLoginScreen.email');
    const fieldPassword = getByPlaceholderText('emailLoginScreen.password');
    fireEvent.changeText(fieldEmail, email);
    fireEvent.changeText(fieldPassword, '');
    fireEvent.press(buttonLogin);

    // wait for event triggered by Formik
    await sleep(1000);
    const error = getByText('error.required');

    expect(authService.signInWithEmailAndPassword).not.toBeCalled();
    expect(error).toBeDefined();
  });
});
