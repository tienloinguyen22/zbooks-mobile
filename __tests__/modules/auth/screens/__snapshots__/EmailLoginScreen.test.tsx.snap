// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/auth/screen/EmailLoginScreen renders successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView
      style={
        Array [
          Object {
            "flex": 0,
          },
          Object {
            "backgroundColor": "black",
          },
        ]
      }
    />
    <SafeAreaView
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "black",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "height": 50,
            },
            Object {
              "backgroundColor": "black",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 20,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "width": 50,
                }
              }
              transparent={true}
            >
              <Text
                allowFontScaling={false}
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontSize": 25,
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 1,
              },
              Object {
                "minWidth": 150,
              },
            ]
          }
        >
          <Text
            bold={true}
            h6={true}
            style={
              Object {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "bold",
                "textTransform": "none",
              }
            }
            white={true}
          >
            emailLoginScreen.login
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 1,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "flex": 0,
                "paddingTop": 10,
              },
              Object {
                "backgroundColor": "black",
              },
            ]
          }
        >
          <View
            center={true}
            row={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "textTransform": "none",
                  },
                  Object {
                    "marginBottom": 5,
                    "marginTop": 5,
                  },
                  Object {
                    "color": "white",
                  },
                ]
              }
            >
              emailLoginScreen.email
            </Text>
          </View>
          <View
            style={
              Array [
                Object {},
                Object {
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "fontSize": 15,
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "borderColor": "black",
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              placeholder="emailLoginScreen.email"
              placeholderTextColor="white"
              rejectResponderTermination={true}
              style={
                Object {
                  "color": "white",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {},
              Object {
                "flex": 0,
                "paddingTop": 10,
              },
              Object {
                "backgroundColor": "black",
              },
            ]
          }
        >
          <View
            center={true}
            row={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "textTransform": "none",
                  },
                  Object {
                    "marginBottom": 5,
                    "marginTop": 5,
                  },
                  Object {
                    "color": "white",
                  },
                ]
              }
            >
              emailLoginScreen.password
            </Text>
            <View
              collapsable={false}
            >
              <TouchableOpacity
                activeOpacity={1}
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 14,
                      },
                      Object {
                        "margin": 8,
                      },
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </TouchableOpacity>
              <Modal
                animationType="fade"
                hardwareAccelerated={false}
                transparent={true}
                visible={false}
              >
                <TouchableOpacity
                  activeOpacity={1}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "flex": 1,
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "height": 0,
                          "left": 0,
                          "overflow": "visible",
                          "position": "absolute",
                          "top": 0,
                          "width": 0,
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        style={
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 14,
                            },
                            Object {
                              "margin": 8,
                            },
                            Object {
                              "fontFamily": "Material Design Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                      >
                        
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "left": -7.5,
                          "position": "absolute",
                          "top": -2,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "borderBottomColor": "white",
                              "borderBottomWidth": 15,
                              "borderLeftColor": "transparent",
                              "borderLeftWidth": 8,
                              "borderRightColor": "transparent",
                              "borderRightWidth": 8,
                              "borderStyle": "solid",
                              "height": 0,
                              "width": 0,
                            },
                            Object {
                              "borderBottomColor": "#fff",
                            },
                            Object {},
                          ]
                        }
                      />
                    </View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#fff",
                          "borderRadius": 10,
                          "display": "flex",
                          "flex": 1,
                          "height": 140,
                          "justifyContent": "flex-start",
                          "left": -17.999,
                          "padding": 10,
                          "position": "absolute",
                          "top": 10,
                          "width": 250,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#222C44",
                            "textTransform": "none",
                          }
                        }
                      >
                        common.passwordInfo
                      </Text>
                    </View>
                  </View>
                </TouchableOpacity>
              </Modal>
            </View>
          </View>
          <View
            style={
              Array [
                Object {},
                Object {
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "fontSize": 15,
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "borderColor": "black",
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              placeholder="emailLoginScreen.password"
              placeholderTextColor="white"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Object {
                  "color": "white",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View
            disabled={false}
            style={
              Array [
                Object {},
                Object {
                  "alignItems": "center",
                  "backgroundColor": "black",
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "width": 200,
                },
                Object {
                  "alignSelf": "center",
                  "marginBottom": 20,
                  "width": 300,
                },
                Object {
                  "marginTop": 20,
                },
                Object {
                  "borderRadius": 20,
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#fff",
                  "fontWeight": "bold",
                  "textTransform": "none",
                }
              }
              white={true}
            >
              emailLoginScreen.login
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View
            disabled={false}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "black",
                "borderRadius": 20,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "marginBottom": 20,
                "width": 300,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#fff",
                  "fontWeight": "bold",
                  "textTransform": "none",
                }
              }
              white={true}
            >
              emailLoginScreen.forgotPassword
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  </View>
</View>
`;

exports[`modules/auth/screen/EmailLoginScreen renders successfully with Vietnamese language 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <SafeAreaView
      style={
        Array [
          Object {
            "flex": 0,
          },
          Object {
            "backgroundColor": "black",
          },
        ]
      }
    />
    <SafeAreaView
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "backgroundColor": "black",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {
              "alignSelf": "flex-start",
              "flexDirection": "row",
              "height": 50,
            },
            Object {
              "backgroundColor": "black",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 20,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "width": 50,
                }
              }
              transparent={true}
            >
              <Text
                allowFontScaling={false}
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#fff",
                      "fontSize": 25,
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
          </TouchableOpacity>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 1,
              },
              Object {
                "minWidth": 150,
              },
            ]
          }
        >
          <Text
            bold={true}
            h6={true}
            style={
              Object {
                "color": "#fff",
                "fontSize": 18,
                "fontWeight": "bold",
                "textTransform": "none",
              }
            }
            white={true}
          >
            emailLoginScreen.login
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 1,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "flex": 0,
                "paddingTop": 10,
              },
              Object {
                "backgroundColor": "black",
              },
            ]
          }
        >
          <View
            center={true}
            row={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "textTransform": "none",
                  },
                  Object {
                    "marginBottom": 5,
                    "marginTop": 5,
                  },
                  Object {
                    "color": "white",
                  },
                ]
              }
            >
              emailLoginScreen.email
            </Text>
          </View>
          <View
            style={
              Array [
                Object {},
                Object {
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "fontSize": 15,
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "borderColor": "black",
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              placeholder="emailLoginScreen.email"
              placeholderTextColor="white"
              rejectResponderTermination={true}
              style={
                Object {
                  "color": "white",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {},
              Object {
                "flex": 0,
                "paddingTop": 10,
              },
              Object {
                "backgroundColor": "black",
              },
            ]
          }
        >
          <View
            center={true}
            row={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "black",
                "flexDirection": "row",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "textTransform": "none",
                  },
                  Object {
                    "marginBottom": 5,
                    "marginTop": 5,
                  },
                  Object {
                    "color": "white",
                  },
                ]
              }
            >
              emailLoginScreen.password
            </Text>
            <View
              collapsable={false}
            >
              <TouchableOpacity
                activeOpacity={1}
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 14,
                      },
                      Object {
                        "margin": 8,
                      },
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  
                </Text>
              </TouchableOpacity>
              <Modal
                animationType="fade"
                hardwareAccelerated={false}
                transparent={true}
                visible={false}
              >
                <TouchableOpacity
                  activeOpacity={1}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "flex": 1,
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "height": 0,
                          "left": 0,
                          "overflow": "visible",
                          "position": "absolute",
                          "top": 0,
                          "width": 0,
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        style={
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 14,
                            },
                            Object {
                              "margin": 8,
                            },
                            Object {
                              "fontFamily": "Material Design Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                      >
                        
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "left": -7.5,
                          "position": "absolute",
                          "top": -2,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "borderBottomColor": "white",
                              "borderBottomWidth": 15,
                              "borderLeftColor": "transparent",
                              "borderLeftWidth": 8,
                              "borderRightColor": "transparent",
                              "borderRightWidth": 8,
                              "borderStyle": "solid",
                              "height": 0,
                              "width": 0,
                            },
                            Object {
                              "borderBottomColor": "#fff",
                            },
                            Object {},
                          ]
                        }
                      />
                    </View>
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#fff",
                          "borderRadius": 10,
                          "display": "flex",
                          "flex": 1,
                          "height": 100,
                          "justifyContent": "flex-start",
                          "left": -17.999,
                          "padding": 10,
                          "position": "absolute",
                          "top": 10,
                          "width": 250,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#222C44",
                            "textTransform": "none",
                          }
                        }
                      >
                        common.passwordInfo
                      </Text>
                    </View>
                  </View>
                </TouchableOpacity>
              </Modal>
            </View>
          </View>
          <View
            style={
              Array [
                Object {},
                Object {
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "fontSize": 15,
                  "paddingBottom": 10,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 10,
                },
                Object {
                  "borderColor": "black",
                },
              ]
            }
          >
            <TextInput
              allowFontScaling={true}
              placeholder="emailLoginScreen.password"
              placeholderTextColor="white"
              rejectResponderTermination={true}
              secureTextEntry={true}
              style={
                Object {
                  "color": "white",
                }
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View
            disabled={false}
            style={
              Array [
                Object {},
                Object {
                  "alignItems": "center",
                  "backgroundColor": "black",
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "width": 200,
                },
                Object {
                  "alignSelf": "center",
                  "marginBottom": 20,
                  "width": 300,
                },
                Object {
                  "marginTop": 20,
                },
                Object {
                  "borderRadius": 20,
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#fff",
                  "fontWeight": "bold",
                  "textTransform": "none",
                }
              }
              white={true}
            >
              emailLoginScreen.login
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View
            disabled={false}
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "black",
                "borderRadius": 20,
                "flexDirection": "row",
                "height": 45,
                "justifyContent": "center",
                "marginBottom": 20,
                "width": 300,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "color": "#fff",
                  "fontWeight": "bold",
                  "textTransform": "none",
                }
              }
              white={true}
            >
              emailLoginScreen.forgotPassword
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  </View>
</View>
`;
